.github/workflows/gradleBuild-component.yml

```diff
@@ -13,31 +13,27 @@ on:
 jobs:
   build_component:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: ${{ inputs.workingDirectory }} 
     steps:
       - uses: actions/checkout@v3
       - name: Setup JDK
         uses: actions/setup-java@v3
         with: 
-          java-version: '18'
+          java-version: 17
           distribution: 'adopt'
           cache: gradle
       - name: Detect Secrets
         uses: RobertFischer/detect-secrets-action@v2.0.0
-      - name: Grant Execute Permission for Gradlew
-        run: chmod +x gradlew
       - name: Build with Gradle
-        run: ./gradlew build
+        run: ./gradlew build -p ${{ inputs.workingDirectory }} 
       - name: Upload Unit Test Results
         if: ${{ always() }}
         uses: actions/upload-artifact@v3
         with:
           name: Unit Tests
           path: |
-              build/reports/tests/test/**
-              build-test-results/**
+              components/**/build/reports/tests/test/**
+              components/**build-test-results/**
+      # TODO: not sure what to do with the next step
       - name: Upload a Build Artifact
```

Comment:
what's collected below probably needs to be updated since we're not setting the working directory anymore

Reply:
As commented above, I'm proposing to change this upload to only upload test results.

Reply:
Looking again at this file and I think that the original intention was to upload the built artifacts, hence proposing this change:
https://github.com/microsoft/kiota-java/pull/118/commits/459c804b346e73ec7ae7984a5da09275e5bd0790

Gradle-related files should be irrelevant.

Reply:
as mentioned in the other comment, we upload 3 things:
- android api level linting report (separate workflow)
- build artifacts
- unit test results
- (some of the sources, but I don't think it was any use and you just removed that part rightfully)

Reply:
Changed to upload all of the build artifacts: `./**/build/*`, build results are explicitly and separately uploaded in the step above.